drivers/sensor/lis2de12/lis2de12_trigger.c

```diff
@@ -0,0 +1,197 @@
+/* ST Microelectronics LIS2DE12 3-axis accelerometer sensor driver
+ *
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lis2de12.pdf
+ */
+
+#define DT_DRV_COMPAT st_lis2de12
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/logging/log.h>
+
+#include "lis2de12.h"
+
+LOG_MODULE_DECLARE(LIS2DE12, CONFIG_SENSOR_LOG_LEVEL);
+
+/**
+ * lis2de12_enable_xl_int - XL enable selected int pin to generate interrupt
+ */
+static int lis2de12_enable_xl_int(const struct device *dev, int enable)
+{
+	const struct lis2de12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lis2de12_ctrl_reg3_t val = {0};
```

Comment:
I don't think you need to initialize this

Reply:
usually it is not needed, but in the lis2de12 stmesmc driver this API is like this

```
int32_t lis2de12_pin_int1_config_set(stmdev_ctx_t *ctx,
                                     lis2de12_ctrl_reg3_t *val)
{
  int32_t ret; 

  ret = lis2de12_write_reg(ctx, LIS2DE12_CTRL_REG3, (uint8_t *) val, 1);

  return ret; 
}
```